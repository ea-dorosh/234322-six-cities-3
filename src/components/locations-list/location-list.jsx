import React from 'react';
import PropTypes from 'prop-types';
import LocationsListItem from "../locations-list-item/location-list-item.jsx";
import {connect} from "react-redux";

const LocationsList = (props) => {
  const {cities} = props;

  return (
    <ul className="locations__list tabs__list">
      {cities.map((city) => (
        <LocationsListItem
          key={city.name}
          city={city}
        />
      ))}
    </ul>
  );
};

LocationsList.propTypes = {
  cities: PropTypes.array.isRequired,
};


const mapStateToProps = (state) => {
  return {
    cities: state.cities,
  };
};


export default connect(mapStateToProps)(LocationsList);
