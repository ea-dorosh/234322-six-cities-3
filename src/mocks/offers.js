import {getRandomNumber, getRandomElement} from "../utils";

const PRICE = {
  MIN: 50,
  MAX: 500
};

const offersNames = [
  `Beautiful & luxurious apartment at great location`,
  `Nice, cozy, wood and stone place`,
  `Wood and stone place`,
  `Nice, cozy, warm big bed apartment`
];

const holderNames = [
  `Angelina`,
  `Max`,
];

const offersTypes = [
  `apartment`,
  `room`,
  `house`,
  `hotel`
];

const offersCords = [
  [52.3909553943508, 4.85309666406198],
  [52.369553943508, 4.85309666406198],
  [52.3909553943508, 4.929309666406198],
  [52.3809553943508, 4.939309666406198]
];

const createOffer = (index) => {
  const bedRoomQuantity = getRandomNumber(1, 6);
  const maxGuestQuantity = bedRoomQuantity * 2;
  const holderName = getRandomElement(holderNames);

  return {
    img: `img/apartment-0${getRandomNumber(1, 3)}.jpg`,
    photos: [
      `img/apartment-0${getRandomNumber(1, 3)}.jpg`,
      `img/apartment-0${getRandomNumber(1, 3)}.jpg`,
      `img/apartment-0${getRandomNumber(1, 3)}.jpg`,
      `img/apartment-0${getRandomNumber(1, 3)}.jpg`,
      `img/apartment-0${getRandomNumber(1, 3)}.jpg`,
      `img/apartment-0${getRandomNumber(1, 3)}.jpg`,
    ],
    isPremium: !!(index % 2),
    price: getRandomNumber(PRICE.MIN, PRICE.MAX),
    name: offersNames[index],
    type: getRandomElement(offersTypes),
    rating: parseFloat(`${getRandomNumber(2, 4)}.${getRandomNumber(0, 9)}`),
    description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    bedRoomQuantity,
    maxGuestQuantity,
    details: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    holder: {
      holderName,
      img: `img/avatar-${holderName.toLowerCase()}.jpg`,
      isSuper: !(index % 2),
    },
    coords: {
      x: offersCords[index][0],
      y: offersCords[index][1],
    }
  };
};

const createOffers = (amount) => {
  const offers = [];

  for (let i = 0; i < amount; i++) {
    const offer = createOffer(i);

    offers.push(offer);
  }
  return offers;

};


export {createOffers};
