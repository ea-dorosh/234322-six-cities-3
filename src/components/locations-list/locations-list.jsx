import React from 'react';
import PropTypes from 'prop-types';
import LocationsListItem from "../locations-list-item/locations-list-item.jsx";
import {connect} from "react-redux";
import {getCitiesList} from "../../reducer/offers/selectors";

export const LocationsList = (props) => {
  const {cities} = props;

  return (
    <ul className="locations__list tabs__list">
      {cities.map((city, index) => (
        <LocationsListItem
          key={index}
          city={city}
        />
      ))}
    </ul>
  );
};

LocationsList.propTypes = {
  cities: PropTypes.array.isRequired,
};


const mapStateToProps = (state) => {
  return {
    cities: getCitiesList(state),
  };
};


export default connect(mapStateToProps)(LocationsList);
