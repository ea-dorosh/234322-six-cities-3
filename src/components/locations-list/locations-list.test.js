import React from "react";
import renderer from "react-test-renderer";
import {LocationsList} from "./locations-list.jsx";
import {citiesMock} from "../../mocks/offers_for_test";

it(`Should LocationsList render correctly`, () => {
  const tree = renderer
    .create(
        <LocationsList
          cities={citiesMock}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
