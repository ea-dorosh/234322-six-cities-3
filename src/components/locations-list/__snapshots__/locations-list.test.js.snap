// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should LocationsList render correctly 1`] = `
<ul
  className="locations__list tabs__list"
>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item "
      href="#"
      onClick={[Function]}
    >
      <span />
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item "
      href="#"
      onClick={[Function]}
    >
      <span />
    </a>
  </li>
</ul>
`;
